Automattic/woocommerce-payments
# MCCY: Round to lowest denomination before applying price rounding from MCCY settings

Fixes #10009

## Changes proposed in this Pull Request









This is consistent with the results of the discussions in paJDYF-g0K-p2.

Previous behavior as of #9876 is to ceil the price before applying any rounding options from the settings, but based on the P2 discussions we identified that using `round` instead of `ceil` is the better approach.

## Testing instructions





1. Make sure unit tests pass.

Then;

> [!NOTE]
> You must have the WooCommerce Deposits plugin active and enabled.

1. Go to **WooCommerce &rarr; Settings &rarr; Multi-Currency** and pick a currency that is not the store currency.
3. Open the settings for that currency and set the manual rate for the currency to `3.783325`.
4. Set "charm pricing" to `none` and "price rounding" to `0.00`.
5. Make sure to enable "Add a currency switcher to the Storefront theme on breadcrumb section."
6. Create a deposits product:
    * Create a simple product. Set the price to $25
    * Go to the "Deposits" tab on the admin product page.
    * Enable deposits with "Yes - deposits are optional"
    * Set Deposit Type to "Inherit storewide settings (percent)"
    * Deposit Amount: 50
    * Save
7. Go to the page of the product you just created.
8. Make sure the selected currency is the one you set the manual rate for in step (2).
9. Make sure the product price is `94.58`
10. Verify "Pay Deposit" is selected.
11. Increase the number to "2" in the quantity input.
12. Click "Add to cart"
13. Make sure the subtotal in the cart is `94.58`.
    * You did not read the price wrong; we're paying 50% deposit on 2 products so we expect this to be the same amount as the single product price :)

Finally;

* Exploratory testing:
    * Play around with a couple different product types, taxes, and shipping costs.
    * Make sure amounts are consistent across product pages, mini-cart, and shortcode/blocks cart and checkout pages.
    * Try at least 2 different non-store currencies, 1 that has 2 decimals, and another that has 0 decimals, e.g. `EUR` and `JPY`, and make sure rounding is consistently applied.
    * Try some combinations of charm pricing and rounding in the currency settings, including with one or both disabled, and make sure prices are rounded correctly.

-------------------


**Post merge**
